mobx_codegen/test/data/valid_annotated_store_output.dart

```diff
@@ -0,0 +1,86 @@
+class User extends _User {
```

Comment:
Need to check if `@JsonSerializable` can be used on the private class, since we won't have access to this public one

Reply:
Would you like me to do that, or are you checking?

Reply:
Don't worry about it. I'll check it.

Reply:
👍 We're good. `@JsonSerializable` works fine on a private class.

Reply:
How do the generated methods look for `@JsonSerializable`? Can they be used from the public facing class instance ? Curious to see some of that code. This is an important use case for many of our users so want to make sure we are not missing it.

Reply:
https://github.com/shyndman/private_json_serializable/blob/master/lib/example.dart

Reply:
Nice! 👍

Reply:
Yeah, they're a _little_ strange (lots of underscores), but imo that's fine. This is deep in implementation-land, and I don't think people will mind.

Reply:
I think we should include one test for `@JsonSerializable` as well. This will be useful in the future as well when we discover newer approaches to doing codegen. Please add one.

Reply:
k